src/odemis/driver/tescan.py

```diff
@@ -1034,25 +1034,42 @@ def _waitSync(self):
 class Stage(model.Actuator):
     """
     This is an extension of the model.Actuator class. It provides functions for
-    moving the Tescan stage and updating the position. 
+    moving the SEM stage with the Tescan Essence software.
     """
     def __init__(self, name, role, parent, **kwargs):
-        """
-        axes (set of string): names of the axes
-        """
-        axes_def = {}
         self._position = {}
-
-        rng = [-0.5, 0.5]
-        # TODO: use StgGetLimits and StgGetMotorized
-        axes_def["x"] = model.Axis(unit="m", range=rng)
-        axes_def["y"] = model.Axis(unit="m", range=rng)
-        axes_def["z"] = model.Axis(unit="m", range=rng)
-        # TODO: support all the 5 axes (also rz and rx)
+        axes_def = {}
+        # limits are always returned as a list with min and max values in the order of x,y,z,r,t
+        # the list should always return 10 values and if an axis is not motorized, limits are zero
+        axes_rng = parent._device.StgGetLimits(1)  # using the soft limits parameter here
+        axes_motorized = parent._device.StgGetMotorized()
+
+        for num, ax in enumerate(["x", "y", "z", "rz", "rx"]):
+            # axes ranges are always returned in pairs of 2 values (min, max)
+            if len(axes_rng) >= 2:
+                # if an axis is not motorized do not add it to the axes definition
+                if axes_motorized[num]:
+                    # make a distinction between linear axes and rotational ones
+                    # rotational axes will need their range converted to radians
+                    if ax.startswith("r"):
+                        axes_rng[0] = 0 if axes_rng[0] == -360 and axes_rng[1] == 360 else axes_rng[0]
```

Comment:
I think this would be more readable (but untested!):
```suggestion
                        if axes_rng[:2] == [-360, 360]:  # Full rotation: report as 0 -> 360°
                             axes_rng[0] = 0
```